pkg/prometheus/promcfg.go

```diff
@@ -1802,5 +1816,9 @@ func (l *limitEnforcer) addLimitsToYAML(cfg yaml.MapSlice, k limitKey, limit uin
 		return cfg
 	}
 
-	return append(cfg, yaml.MapItem{Key: k.prometheusField, Value: getLimit(limit, enforcedLimit)})
+	// Since BodySizeLimit is defined only in PrometheusCRD
+	if k.specField != "enforcedBodySizeLimit" {
```

Comment:
we don't need a special case for this scenario since `getLimit` will handle the fact that the BodySizeLimit is only defined at the PrometheusCRD level.

Reply:
Ya it was my bad I read the conditional in getLimits() wrong. I updated the pR